src/solace_ai_connector/components/general/db/mongo/mongo_search.py

```diff
@@ -0,0 +1,97 @@
+"""MongoDB Agent Component for handling database search."""
+import json
+
+from .mongo_handler import MongoHandler
+from ....component_base import ComponentBase
+
+
+info = {
+    "class_name": "MongoDBSearchComponent",
+    "description": "Searches a MongoDB database.",
+    "config_parameters": [
+        {
+            "name": "database_host",
+            "required": True,
+            "description": "MongoDB host",
+            "type": "string",
+        },
+        {
+            "name": "database_port",
+            "required": True,
+            "description": "MongoDB port",
+            "type": "integer",
+        },
+        {
+            "name": "database_user",
+            "required": False,
+            "description": "MongoDB user",
+            "type": "string",
+        },
+        {
+            "name": "database_password",
+            "required": False,
+            "description": "MongoDB password",
+            "type": "string",
+        },
+        {
+            "name": "database_name",
+            "required": True,
+            "description": "Database name",
+            "type": "string",
+        },
+        {
+            "name": "database_collection",
+            "required": False,
+            "description": "Collection name - if not provided, all collections will be used",
+        },
+    ],
+        "input_schema": {
+        "type": "object",
+        "properties": {
+            "collection": {
+                "type": "string",
+                "description": "The collection to search in.",
+            },
+            "query": {
+                "type": "object",
+                "description": "The query pipeline to execute. if string is provided, it will be converted to JSON.",
+            }
+        },
+    },
+}
+
+
+class MongoDBSearchComponent(ComponentBase):
+    """Component for handling MongoDB database operations."""
+
+    def __init__(self, **kwargs):
+        """Initialize the MongoDB component.
+
+        Args:
+            **kwargs: Additional keyword arguments.
+
+        Raises:
+            ValueError: If required database configuration is missing.
+        """
+        super().__init__(info, **kwargs)
+
+        # Initialize MongoDB handler
+        self.db_handler = MongoHandler(
+            self.get_config("database_host"),
+            self.get_config("database_port"),
+            self.get_config("database_user"),
+            self.get_config("database_password"),
+            self.get_config("database_collection"),
+            self.get_config("database_name"),
+        )
+
+    def invoke(self, message, data):
+        collection = data.get("collection")
+        query = data.get("query")
+        if not query:
+            raise ValueError("No query provided")
+        if not collection:
+            raise ValueError("No collection provided")
```

Comment:
is the collection is missing in the payload, should not it use the self.collection as it is encoded in the db_handler.execute_query?

Reply:
Good point. updated